src/snackbar/SnackbarTraceDialog.tsx

```diff
@@ -0,0 +1,69 @@
+import { Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, withMobileDialog } from "@material-ui/core";
+import * as React from "react";
+import { SnackbarManager_Message_t } from "./SnackbarManager";
+import { SnackbarTraceHeaderConsumer } from "./SnackbarTraceHeaderProvider";
+
+type SnackbarTraceDialog_Props_t = {
+    message: SnackbarManager_Message_t,
+    open: boolean,
+    fullScreen?: boolean,
+    onClose: () => void,
+};
+
+function createTraceMessage(message: SnackbarManager_Message_t, headers: { [k: string]: string }): string {
+    const messages = [
+        "-----BEGIN FINDER TRACE ----",
+        "Type: " + message.type,
+        "Message: " + message.message,
+    ];
+    for (const header of Object.keys(headers)) {
+        messages.push(header + ": " + headers[header]);
+    }
+    if (message.trace) {
+        messages.push("");
+        messages.push(message.trace);
+    }
+    messages.push("-----END FINDER TRACE-----");
+
+    return messages.join("\n");
+}
+
+function SnackbarTraceDialog({ message, onClose, ...props }: SnackbarTraceDialog_Props_t) {
+    const preRef = React.createRef<HTMLPreElement>();
+    return <Dialog
+        {...props}
+        scroll="paper"
+        onClose={() => onClose()}
+        fullWidth={true}
+        maxWidth={false}
+    >
+        <DialogTitle>Trace: {message.message}</DialogTitle>
+        <DialogContent>
+            <DialogContentText>
+                <SnackbarTraceHeaderConsumer>{
+                    (headers) =>
+                        <pre ref={preRef}>{createTraceMessage(message, headers)}</pre>
```

Comment:
These headers will have to be provided by the application that embeds this component.

Because we don't want to pass that down as a separate property all the way through the snackbar components, we pass it as a react context value: https://reactjs.org/docs/context.html

Since it is data that does not change in a running application, context is a nice way to pass it through from the top level to the consumers in a lower level.

The default value is set here. It is an empty object in this case.
 https://github.com/xenit-eu/finder-ui/blob/574156310e37b9b16a134fc88963fafc683d63f4/src/snackbar/SnackbarTraceHeaderProvider.tsx#L3